cmd/lotus-miner/market.go

```diff
@@ -910,6 +911,34 @@ var dealsPendingPublish = &cli.Command{
 	},
 }
 
+var dealsRetryPublish = &cli.Command{
+	Name:  "retry-publish",
+	Usage: "retry publishing a deal",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name: "proposal-cid",
+		},
+	},
```

Comment:
Maybe simpler to make this a parameter rather than a flag.
```
lotus-miner storage-deals retry-publish <proposal cid>
```
instead of
```
lotus-miner storage-deals retry-publish --proposal-cid=<proposal cid>
```